packages/spor-button-react-native/src/Button.tsx

```diff
@@ -7,69 +7,156 @@ import {
+});
+
+type ButtonVariant =
+  | "primary"
+  | "control"
+  | "secondary"
+  | "tertiary"
+  | "additional";
 
 type RestyleProps = SpacingProps<Theme> &
   VariantProps<Theme, "buttonVariants", "variant"> &
   VariantProps<Theme, "buttonSizes", "size">;
 
 const sizes = createVariant({ themeKey: "buttonSizes", property: "size" });
-const variants = createVariant({
-  themeKey: "buttonVariants",
-  property: "variant",
-});
 
 const restyleFunctions = composeRestyleFunctions<Theme, RestyleProps>([
   spacing,
   sizes,
   variants,
 ]);
 
-type Props = RestyleProps & {
+const ANDROID_VERSION_LOLLIPOP = 21;
+const ANDROID_SUPPORTS_RIPPLE =
+  Platform.OS === "android" && Platform.Version >= ANDROID_VERSION_LOLLIPOP;
+
+type ButtonProps = Exclude<RestyleProps, "variant"> & {
+  variant: ButtonVariant;
   onPress: () => void;
-  children: React.ReactNode;
-  isDisabled?: boolean;
+  children: string;
+  disabled?: boolean;
   isLoading?: boolean;
-  leftIcon?: React.ReactNode;
-  rightIcon?: React.ReactNode;
+  accessibilityLabel?: string;
 };
 
 /** A button. */
 export const Button = ({
+  variant,
   onPress,
-  children,
-  isDisabled = false,
+  disabled = false,
   isLoading = false,
-  leftIcon,
-  rightIcon,
+  accessibilityLabel,
+  children,
   ...rest
-}: Props) => {
-  if (isDisabled) {
-    rest.variant = "disabled";
-  }
-  const { style, ...viewProps } = useRestyle(restyleFunctions, rest);
+}: ButtonProps) => {
+  const [pressed, setPressed] = useState(false);
 
-  const fontSize = findProperty("fontSize", style as any[]);
-  const fontWeight = findProperty("fontWeight", style as any[]);
-  const color = findProperty("color", style as any[]);
+  const { style: restyleStyle } = useRestyle(restyleFunctions, {
+    variant,
+    ...rest,
+  });
+  const activeStyle = pressed ? getActiveStyle(variant) : {};
+  const disabledAndLoadingStyle =
+    disabled || isLoading ? getDisabledAndLoadingStyle() : {};
+
+  const style = [
+    restyleStyle,
+    activeStyle,
+    disabledAndLoadingStyle,
+  ] as StyleProp<TextStyle>;
+
+  const flatStyles = StyleSheet.flatten(style);
+  const fontSize = flatStyles.fontSize;
+  const fontWeight = flatStyles.fontWeight;
+  const color = flatStyles.color;
+  const backgroundColor = flatStyles.backgroundColor;
 
   return (
-    <TouchableOpacity onPress={onPress} disabled={isDisabled}>
-      <View style={style as any} {...viewProps}>
-        {isLoading ? (
-          <ActivityIndicator color={color} />
-        ) : (
-          <Text style={{ color, fontSize, fontWeight }}>{children}</Text>
-        )}
-      </View>
-    </TouchableOpacity>
+    <Pressable
+      accessibilityRole="button"
+      accessibilityState={{ disabled }}
+      onPressIn={() => setPressed(true)}
+      onPressOut={() => setPressed(false)}
+      onPress={onPress}
+      disabled={disabled}
+      accessibilityLabel={accessibilityLabel}
+      style={style}
+      android_ripple={
+        ANDROID_SUPPORTS_RIPPLE
```

Comment:
Er det noe i veien for å sende inn `android_ripple` configen uansett?

Reply:
Nei, det bør jo gå fint nå 😄